### ISSUE TITLE
Not working for Stellaris/Cities Skylines

### ISSUE THREAD

**Issue by DoodleSchrank** [20PercentRendered/creamlinux#10]
- Games are owned on Steam
- I followed the install instructions
- added the correct DLC lines for Cities Skylines
- a little window from cream.sh opens
- I say "play game" or "start launcher"
- DLC are not unlocked.
I'm using Arch (5.19.10) btw.

**Comment by 20PercentRendered** [comment]
Do you have the actual files for the dlcs? Creamlinux just hooks steam calls, it doesn't magically install any actual dlc depots.

**Comment by DoodleSchrank** [comment]
From my knowledge Cities and Stellaris both keep the files on your drive whether or not you bought the DLC.
At least on Windows I was able to easily crack the DLC with creamapi with no further downloads.

**Comment by 20PercentRendered** [comment]
Cities skylines is a unity game. Creamlinux does seem to have some trouble with unity games. I don't know why though. Stellaris _should_ work, but I haven't tested. For now you can use creamapi through proton.

**Comment by 20PercentRendered** [comment]
Please retest stellaris with the new 2.0 creamlinux version.

**Comment by 20PercentRendered** [comment]
I've heard that some version of Stellaris added additional server-side checks. I might look into this in the future if I decide to buy it or if paradox implements similar checks for hoi4. But for now you can use the game through proton and use whatever you would normally use.

**Comment by clamor-s** [comment]
@20PercentRendered thanks!

**Comment by 20PercentRendered** [comment]
I am currently working on adding unity support for creamlinux. I already added dlsym support and that made GWYF work, but Cities Skylines's paradox launcher seems to break due to dlsym. Oddly enough, it doesn't affect hoi4. 

If you want to test bleeding-edge releases, you can go into Actions, then click the newest commit and download linux-artifacts. Note that these builds may have issues supporting some games.

**Comment by clamor-s** [comment]
@20PercentRendered Stellaris got update with some dlc checksum fixes, unfortunately I was not able to test it yet :( will update you asap

**Comment by clamor-s** [comment]
@20PercentRendered I can not build it anymore

**Comment by 20PercentRendered** [comment]
Yeah dlsym hooking definitely broke a lot of things like PAYDAY2 and the paradox launcher. I'm not going to revert the commits yet, as I'm going to try to figure out what's breaking. I might also make two versions of creamlinux, one with dlsym hooking and one without. If you want to, you could comment the dlsym lines out and uncomment the SteamClient lines and then test if that works.

**Comment by 20PercentRendered** [comment]
> @20PercentRendered I have some additional tests with Stellaris. Creamlinux works perfectly fine (and builds as well) with 4 last commits reverted (dlsym hooking) and pre 3.6 game version. It does not with 3.6 and later. Repo on master fails to build for me and artifacts do not work either. If you have some stuff to test just call me and I will test.

Compiling the latest commit works for me, you might be missing some 32-bit libraries. What is the error you are getting on compile?

**Comment by clamor-s** [comment]
@20PercentRendered 
`    /usr/bin/ld: пропускаємо несумісний /usr/lib/gcc/x86_64-linux-gnu/12/libstdc++.so під час пошуку -lstdc++
    /usr/bin/ld: пропускаємо несумісний /usr/lib/gcc/x86_64-linux-gnu/12/libstdc++.a під час пошуку -lstdc++
    /usr/bin/ld: не вдалося знайти -lstdc++: Немає такого файла або каталогу
    /usr/bin/ld: пропускаємо несумісний /usr/lib/gcc/x86_64-linux-gnu/12/libstdc++.so під час пошуку -lstdc++
`

**Comment by 20PercentRendered** [comment]
Looks like you need `gcc-multilib` and `g++-multilib`

**Comment by clamor-s** [comment]
It built flawlessly now

a bit of game log
`[21:25:34][persistent.cpp:46]: Error: "Unexpected token: checksum, near line: 9
" in file: "here is dlc path" near line: 9`

**Comment by 20PercentRendered** [comment]
What is in your cream_api.ini?

**Comment by clamor-s** [comment]
same as in your repo

**Comment by 20PercentRendered** [comment]
That's probably stellaris using unimplemented funcs like SteamClient() or maybe stellaris is asking for some dlc paths through steam. If I ever buy stellaris then I can test. 
But I am testing Cities Skylines right now (as I own it), and it does things even more strangely than GWYF.

**Comment by clamor-s** [comment]
@20PercentRendered may I somehow provide you logs?

**Comment by 20PercentRendered** [comment]
If the game has logs of some sort, you can send those. If it doesn't, run the game with `konsole --hold -e %command%` as a launch option and see if logs appear in Konsole. Unfortunately, creamlinux doesn't contain a file logger. For unity games the path is `~/.config/unity3d/CompanyName/ProductNamePlayer.log`.

**Comment by anticitizn** [comment]
This might be tangential, but several Stellaris DLCs got updated after 3.6 (and seemingly all DLC checksums are generated differently now). Is it possible that the crash is simply caused by outdated DLC files and nothing related to creamlinux? I don't have access to my computer to test right now, but will do within a couple of days.

**Comment by clamor-s** [comment]
@anticitizn Before on windows version I had falloff of herbal dlc after federations. But all other dlc were working fine. Then I switched to @20PercentRendered version and linux, everything worked fine. After 3.6 ALL dlc just fall off with checksum fail.

**Comment by anticitizn** [comment]
@clamor-s That was my experience too, running with DLC files mostly downloaded long ago. Have you tried running with the newest version of all DLCs?

**Comment by clamor-s** [comment]
@clamor-s I could not find latest one. I may look for them later today.

**Comment by clamor-s** [comment]
@20PercentRendered how can I buy you stellaris?

**Comment by rednatnas** [comment]
Working in Stellaris 3.7.3! You need the last DLC's versions since some of them were updated along with the game.
